### ISSUE TITLE
Using git-cop with Travis

### ISSUE THREAD

**Issue by Ana06** [bkuhlmann/git-cop#12]
From the readme

> Calculation of commits is done by reviewing all commits made on the current feature branch since branching from master. Some CI servers don't respect this and blow away any branch information, most notibly, Travis CI. For that reason, Travis CI is not supported or recommended as they use git clone --depth=<number> cloning which can't be customized and destroys any knowledge of master and feature branch information.

Does that mean that git-cop can not be used in Travis?

**Comment by bkuhlmann** [comment]
I'm afraid so. If it's any consolation, I reached out to the Travis CI support team prior to the 1.0.0 release trying to get them to allow anyone to configure their `.travis.yml` file where the `--depth` option could be removed when doing a `git clone --depth`. They said they'd look into it. While they do allow you to customize the Git cloning process they *do not* allow you to remove the `--depth` option. Due to the `--depth` option being used, feature branch commits can't be calculated properly. 😢

**Comment by bkuhlmann** [comment]
Closing this issue for now. Would like to be able to support Travis CI but we can't proceed until Travis CI adds better Git cloning support.

**Comment by Ana06** [comment]
I think you should close it.

I am already using git-cop in Travis. You can see here two examples (in a local branch and in a PR) and it is working perfectly after some workarounds in Travis: :tada: 

https://travis-ci.org/Ana06/open-build-service/jobs/250334265
https://travis-ci.org/openSUSE/open-build-service/jobs/250334288

I have to finish some things (some test are broken after the changes) but I will send a PR for supporting Travis in the following days.

**Comment by bkuhlmann** [comment]
Oh, nice. Actually, I'll re-open and assign to you. This'll be nice to have.
